frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboardView.tsx

```diff
@@ -0,0 +1,242 @@
+import cx from "classnames";
+import { assoc } from "icepick";
+import type { HandleThunkActionCreator } from "react-redux";
+
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import ColorS from "metabase/css/core/colors.module.css";
+import CS from "metabase/css/core/index.css";
+import DashboardS from "metabase/css/dashboard.module.css";
+import type {
+  setParameterValueToDefault as setParameterValueToDefaultDashboardAction,
+  setParameterValue as setParameterValueDashboardAction,
+} from "metabase/dashboard/actions";
+import type { NavigateToNewCardFromDashboardOpts } from "metabase/dashboard/components/DashCard/types";
+import { DashboardEmptyStateWithoutAddPrompt } from "metabase/dashboard/components/Dashboard/DashboardEmptyState/DashboardEmptyState";
+import { getDashboardActions } from "metabase/dashboard/components/DashboardActions";
+import { DashboardGridConnected } from "metabase/dashboard/components/DashboardGrid";
+import { DashboardTabs } from "metabase/dashboard/components/DashboardTabs";
+import type {
+  DashboardFullscreenControls,
+  DashboardRefreshPeriodControls,
+  EmbedHideDownloadButton,
+  EmbedHideParameters,
+  EmbedThemeControls,
+  RefreshPeriod,
+} from "metabase/dashboard/types";
+import { isActionDashCard } from "metabase/dashboard/utils";
+import { isWithinIframe } from "metabase/lib/dom";
+import ParametersS from "metabase/parameters/components/ParameterValueWidget.module.css";
+import type { DisplayTheme } from "metabase/public/lib/types";
+import { EmbeddingSdkMode } from "metabase/visualizations/click-actions/modes/EmbeddingSdkMode";
+import { PublicMode } from "metabase/visualizations/click-actions/modes/PublicMode";
+import type { UiParameter } from "metabase-lib/v1/parameters/types";
+import type {
+  Dashboard,
+  DashboardCard,
+  DashboardId,
+  ParameterValueOrArray,
+} from "metabase-types/api";
+import type { SelectedTabId } from "metabase-types/store";
+
+import { EmbedFrame } from "../../components/EmbedFrame";
+
+import { DashboardContainer } from "./PublicOrEmbeddedDashboard.styled";
+
+export function PublicOrEmbeddedDashboardView({
+  dashboard,
+  hasNightModeToggle,
+  isFullscreen,
+  isNightMode,
+  onFullscreenChange,
+  onNightModeChange,
+  onRefreshPeriodChange,
+  refreshPeriod,
+  setRefreshElapsedHook,
+  selectedTabId,
+  parameters,
+  parameterValues,
+  draftParameterValues,
+  setParameterValue,
+  setParameterValueToDefault,
+  dashboardId,
+  bordered,
+  titled,
+  theme,
+  hideParameters,
+  hideDownloadButton,
+  navigateToNewCardFromDashboard,
+  slowCards,
+  cardTitled,
+}: {
+  dashboard: Dashboard | null;
```

Comment:
After refactoring, some suggested types here were really weird, so it requires manual correction.